chore: drop named query table

sakce opened the pull request:
Finalizes removing the NamedQuery model in a 2-step process (1st @ #39392)

sakce committed:
chore: drop named query table

sakce committed:
Merge branch 'master' into chore/drop-named-quer

webjunkie commented:
I'm looking into what the problem here is! It seems we have very picky tests that migration forward and backward. But they actually shouldn't care about migrations that happen after the ones that they are testing 🤔

webjunkie committed:
Adjust after master merge

webjunkie committed:
Merge branch 'master' into chore/drop-named-quer

# Conflicts:
#	posthog/migrations/max_migration.txt

webjunkie committed:
Try to fix by adjusting tear down for migration tests

webjunkie committed:
Merge remote-tracking branch 'origin/master' into chore/drop-named-quer

# Conflicts:
#	posthog/migrations/max_migration.txt

webjunkie committed:
fix(migrations): renumber to 0883 and optimize test teardown

Renumber drop_named_query migration from 0882 to 0883 after master merge added 0881 and 0882.

Update NonAtomicTestMigrations.tearDownClass to skip forward migration since these tests require --create-db for proper database isolation. Forward migration in teardown was causing conflicts when later migrations drop tables that earlier migration tests need to reverse.

webjunkie committed:
fix(migrations): add reversible drop for namedquery table

Migration 0883 drops the posthog_namedquery table. When migration tests reverse through older migrations, they need the table structure to exist so constraints can be properly removed.

Added reverse_sql that recreates minimal table structure with the unique constraint, allowing earlier migrations (like 0860) to reverse cleanly.

webjunkie committed:
fix(migrations): renumber to 0885 after master merge

Master added migrations 0883 and 0884, so renumbering drop_named_query from 0883 to 0885 and updating dependency to 0884_experimenttimeseriesrecalculation_and_more.

webjunkie commented:
Now another problem. Why do we need to test these old migrations anyways 😩

webjunkie committed:
revert: remove NonAtomicTestMigrations teardown override

Not needed - migration fix alone handles the test failures.

webjunkie committed:
fix(migrations): renumber to 0886 after master merge

Master added 0885_migrate_notebooks_models, renumbering drop_named_query from 0885 to 0886.

sakce commented on posthog/migrations/0886_drop_named_query.py:
```diff
@@ -0,0 +1,21 @@
+                );
+                ALTER TABLE posthog_namedquery ADD CONSTRAINT unique_team_named_query_name UNIQUE (team_id, name);
+            """,
+        ),
```
Nope, wrong Graphite.

sakce committed:
rebase migrations
